nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -141,3 +139,73 @@ def __exit__(self, *args):
 
     def __del__(self):
         self.vc.release()
+
+
+class MovieDataChunkIterator(GenericDataChunkIterator):
+    """DataChunkIterator specifically for use on RecordingExtractor objects."""
+
+    def __init__(
+        self,
+        movie_file: FilePathType,
+        buffer_gb: float = None,
+        chunk_shape: tuple = None,
+        stub_test: bool = False,
+    ):
+        self.video_capture_ob = VideoCaptureContext(movie_file)
+        buffer_gb = 1.0 if buffer_gb is None else buffer_gb
+        self._maxshape = self._get_maxshape()
+        self._full_frame_size_mb, self._full_frame_shape = self._get_frame_details()
+        self._dtype = self._get_dtype()
```

Comment:
```suggestion
        self._full_frame_size_mb, self._full_frame_shape = self._get_frame_details()
```

These are covered in the `super().__init()` call: https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/data_utils.py#L208-L209